code/go/0chain.net/smartcontract/zcnsc/mint.go

```diff
@@ -114,11 +113,43 @@ func (zcn *ZCNSmartContract) Mint(trans *transaction.Transaction, inputData []by
 	// record the global nonce from solidity smart contract
 	gn.WZCNNonceMinted[payload.Nonce] = true
 
+	var (
+		amount currency.Coin
+		n      currency.Coin
+		share  currency.Coin
+	)
+	share, _, err = currency.DistributeCoin(gn.ZCNSConfig.MaxFee, int64(len(payload.Signatures)))
+	if err != nil {
+		err = errors.Wrap(err, fmt.Sprintf("%s, DistributeCoin operation, %s", code, info))
+		return
+	}
+	n, err = currency.Int64ToCoin(int64(len(payload.Signatures)))
+	if err != nil {
+		err = errors.Wrap(err, fmt.Sprintf("%s, convert len signatures to coin, %s", code, info))
+		return
+	}
+	amount, err = currency.MinusCoin(payload.Amount, share*n)
+	if err != nil {
+		err = errors.Wrap(err, fmt.Sprintf("%s, payload.Amount - share * len(signatures), %s", code, info))
+		return
+	}
+	for _, sig := range payload.Signatures {
```

Comment:
checked how we process transfers [here](https://github.com/0chain/0chain/blob/e28bff708acbc89a5cee677ed082aa184a87756e/code/go/0chain.net/chaincore/chain/state.go#L420:L463)
and looks like if client has no balance transfer won't be possible with ``transfer amount - insufficient balance`` error. 
I think that we simply should mint share also, the same way we mint amount and we are fine